grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -155,33 +195,77 @@ class ProductAvailabilityService {
-                pendingRequisitionStatus    : RequisitionStatus.listPending(),
-                pendingOrderStatus          : OrderStatus.listPending(),
-                product                     : product?.id ?: ''
-        ], [readOnly: true])
-
-        return results.collect { [binLocation: it[0]?.id, inventoryItem: it[1]?.id, quantityAllocated: it[2]] }
+        def query = """
+            SELECT bin_location_id, inventory_item_id, SUM(quantity) FROM (
+                SELECT
+                    pli.bin_location_id as bin_location_id,
+                    pli.inventory_item_id as inventory_item_id,
+                    sum(pli.quantity) as quantity
+                FROM picklist_item pli
+                    INNER JOIN picklist p ON pli.picklist_id = p.id
+                    LEFT JOIN requisition_item ri ON pli.requisition_item_id = ri.id
+                    LEFT JOIN requisition r ON p.requisition_id = r.id
+                    LEFT JOIN inventory_item ii ON pli.inventory_item_id = ii.id
+                    LEFT JOIN location l ON pli.bin_location_id = l.id
+                    LEFT JOIN (
+                        SELECT
+                            GROUP_CONCAT(lsa_select.supported_activities_string) as activities,
+                            lsa_select.location_id as location_id
+                        FROM location_supported_activities lsa_select
+                        GROUP BY lsa_select.location_id
+                    ) lsa ON lsa.location_id = l.id
+                WHERE (r.origin_id = :locationId AND r.status IN (${RequisitionStatus.listPending().collect { "'$it'" }.join(',')}))
+                  AND (ri.id IS NOT NULL AND (ii.lot_status IS NULL OR ii.lot_status != 'RECALLED') AND (lsa.activities IS NULL OR lsa.activities NOT LIKE '%HOLD_STOCK%'))
+                  AND (:productId = '' OR ri.product_id = :productId)
+                GROUP BY pli.bin_location_id, pli.inventory_item_id
+                UNION
+                SELECT
+                    pli.bin_location_id as bin_location_id,
+                    pli.inventory_item_id as inventory_item_id,
+                    sum(pli.quantity) as quantity
+                FROM picklist_item pli
+                    INNER JOIN picklist p ON pli.picklist_id = p.id
+                    LEFT JOIN order_item oi ON pli.order_item_id = oi.id
+                    LEFT JOIN `order` o ON p.order_id = o.id
+                    LEFT JOIN inventory_item ii ON pli.inventory_item_id = ii.id
+                WHERE (o.origin_id = :locationId AND o.status IN (${OrderStatus.listPending().collect { "'$it'" }.join(',')}))
+                  AND (oi.id IS NOT NULL)
+                  AND (:productId = '' OR oi.product_id = :productId)
+                GROUP BY pli.bin_location_id, pli.inventory_item_id
+            ) as requisition_order_union
+            GROUP BY bin_location_id, inventory_item_id;
+        """
+
+        def results = dataService.executeQuery(
```

Comment:
If you can't figure out the issue where, try using Hibernate to execute the query instead of the Groovy SQL client we're using. 

https://dzone.com/articles/grails-goodness-using